src/test/java/org/jenkinsci/plugins/workflow/cps/CpsVmExecutorServiceTest.java

```diff
@@ -160,6 +160,24 @@
         }
     }
 
+    @Issue("JENKINS-58620")
+    @Test public void wrongCatcherGroovyShellEvaluate() throws Exception {
+        boolean origFailOnMismatch = CpsVmExecutorService.FAIL_ON_MISMATCH;
+        CpsVmExecutorService.FAIL_ON_MISMATCH = false;
+        try {
+            WorkflowJob p = r.createProject(WorkflowJob.class, "p");
+            errors.checkSucceeds(() -> {
+                p.setDefinition(new CpsFlowDefinition("org.jenkinsci.plugins.workflow.cps.CpsThread.current().getExecution().getShell().evaluate('echo \"Goofy\"', 'X.groovy')", false));
```

Comment:
I suspect just

```suggestion
                p.setDefinition(new CpsFlowDefinition("evaluate('echo(/Goofy/)', 'X.groovy')", false));
```

would suffice to reproduce.

Reply:
No, this logic will not test the issue. I tried to use some compact commands - but only the one I used here is showing the notice.